### PR TITLE
Add tests for same origin checks in document.open()

### PR DESCRIPTION
**Pull request by jeisinger**
Test for whatwg/html#536 and whatwg/html#2282

### CODE REVIEW

**Comment by jeisinger** [comment]
@zcorpan @annevk ptal

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open.html
Comment (annevk) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open.html]: Maybe add another one here where you open `/common/blank.html` and only set document.domain on this side?

The only problem with document.domain I think is that it's set for the whole document. So once you set it, it's set for all following tests too. So maybe it's better to have separate HTML files for the tests.

**Review by annevk — CHANGES_REQUESTED** [review]

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open.html
Comment (jeisinger) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open.html]: ptal

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-document-domain-not-matching.html
Comment (annevk) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-document-domain-not-matching.html]: I don't think this should throw. Why would it?

**Review by annevk — CHANGES_REQUESTED** [review]

**Comment by annevk** [comment]
@nox these tests might interest you.

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-document-domain-not-matching.html
Comment (jeisinger) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-document-domain-not-matching.html]: the test has document.domain set, while /common/blank.html has not, so the "doc" is not "same origin" which is required by step 4

Comment (annevk) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-document-domain-not-matching.html]: It is same origin. Same origin doesn't look at the domain part of the origin tuple.

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-same-origin-domain-ports.html
Comment (annevk) [inline html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-same-origin-domain-ports.html]: This should say same origin-domain, twice.

**Review by annevk — COMMENTED** [review]
This looks okay otherwise, but we're no longer really testing the comparison.

**Comment by jeisinger** [comment]
done.

I agree that it's unfortunate that we can't test the security check, but as you point out on the html pr, step 4 is redundant with the security check in idl, so I don't know how to write a meaningful test for it.

**Comment by tobie** [comment]
Ping @wpt-pr-bot.

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision 61101f3c68ff4d541215830b1e91458342cb156b
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4642/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-basic.html">/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-basic.html</a></summary>


|                        Subtest                         | Results |
|--------------------------------------------------------|---------|
|                                                        | OK      |
| `It should be possible to open same origin documents.` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4642/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-same-origin-domain-ports.html">/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-same-origin-domain-ports.html</a></summary>


|                                         Subtest                                         | Results |
|-----------------------------------------------------------------------------------------|---------|
|                                                                                         | TIMEOUT |
| `It should be possible to open same origin-domain documents (with non-matching ports).` | TIMEOUT |

</details>
</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision 61101f3c68ff4d541215830b1e91458342cb156b
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4642/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-basic.html">/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-basic.html</a></summary>


|                        Subtest                         | Results |
|--------------------------------------------------------|---------|
|                                                        | OK      |
| `It should be possible to open same origin documents.` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4642/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-same-origin-domain-ports.html">/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open-same-origin-domain-ports.html</a></summary>


|                                         Subtest                                         | Results |
|-----------------------------------------------------------------------------------------|---------|
|                                                                                         | OK      |
| `It should be possible to open same origin-domain documents (with non-matching ports).` | FAIL    |

</details>
</details>

**Review by annevk — APPROVED** [review]
This looks good. I can merge this once I or someone else lands the HTML Standard change.

**Pull request merged**
